src/lib/server/middleware/http-request.ts

```diff
@@ -0,0 +1,65 @@
+import { json, type RequestEvent } from "@sveltejs/kit"
+import { logger } from "@vestfoldfylke/loglady"
+import type { AuthenticatedUser } from "$lib/types/authentication"
+import { getAuthenticatedUser } from "../auth/get-authenticated-user"
+import { HTTPError } from "./http-error"
+
+type MiddlewareNextResponse = {
+	response: Response
+	isAuthorized: boolean
+}
+
+type MiddlewareNextParams = {
+	requestEvent: RequestEvent
+	user: AuthenticatedUser
+}
+
+export type MiddlewareNextFunction = (params: MiddlewareNextParams) => Promise<MiddlewareNextResponse>
+
+/**
+ * Wrap functionality in a server route with this middleware to handle authentication, some simple logging and error handling.
+ * The `next` function will only be called if authentication is successful.
+ * Make sure to handle authorization (access to resources) inside the `next` function and either throw a HTTPError or return `isAuthorized: false` if the caller is not authorized to access the requested resource.
+ *
+ * @param {RequestEvent} requestEvent The incoming request event from SvelteKit server route
+ * @param {MiddlewareNextFunction} next The middleware next function to call if authentication is successful, passing in the request event and authenticated user
+ * @returns {Promise<Response>} The final response to return to the client
+ */
+export const httpRequestMiddleware = async (requestEvent: RequestEvent, next: (params: MiddlewareNextParams) => Promise<MiddlewareNextResponse>): Promise<Response> => {
+	const request = requestEvent.request
+	let loggerPrefix = `[HTTP Request Middleware] - ${request.method} ${request.url}`
+
+	logger.info(`${loggerPrefix} - Incoming request`)
+
+	let user: AuthenticatedUser
+	try {
+		user = getAuthenticatedUser(request.headers)
+		loggerPrefix += ` - User: ${user.userId} (${user.name})`
+		logger.info(`${loggerPrefix} - Authenticated user: {userId}`, user.userId)
+	} catch (err) {
+		const errorMsg = err instanceof Error ? err.message : String(err)
+		logger.error(`${loggerPrefix} - Error during authentication: {error}`, errorMsg)
```

Comment:
Use `logger.errorException` instead. No need to check **err** yourself. loglady does this for us

```suggestion
		logger.errorException(err, `${loggerPrefix} - Error during authentication)
```

Reply:
As far as i know, `axios` is the only one i have encountered that does that